packages/compiler-sfc/src/compileStyle.ts

```diff
@@ -15,6 +9,7 @@ import {
 } from './stylePreprocessors'
 import { RawSourceMap } from 'source-map'
 import { cssVarsPlugin } from './cssVars'
+import LazyResult from 'postcss/lib/lazy-result'
```

Comment:
Since PostCSS 8.1.8, you can replace `LazyResult` with `Promise<Result>` and avoid relying on postcss internals ( [changelog](https://github.com/postcss/postcss/blob/master/CHANGELOG.md))

Reply:
Thanks for the hint, but this means i need to drop those lines too (line 183-186)
```javascript
recordPlainCssDependencies(result.messages)
// force synchronous transform (we know we only have sync plugins)
code = result.css
outMap = result.map
```

Reply:
Ah, yes, indeed. `css` on `LazyResult` is meant to be for debug purposes. Since you cannot get rid of promises entirely, I wonder why they try to force the sync processing, it seems to make the code more complicated. You need to return

```typescript
SFCStyleCompileResults | Promise<SFCStyleCompileResults>
```
instead of just

```
Promise<SFCStyleCompileResults>
```

Reply:
It seems we can't get rid of `LazyResult`. 
```javascript 
export interface SFCStyleCompileResults {
  code: string
  map: RawSourceMap | undefined
  rawResult: Promise<Result> | Result | undefined
  errors: Error[]
  modules?: Record<string, string>
  dependencies: Set<string>
}

...

let result:  Promise<Result> | Result | undefined

...

try {
    // In async mode, return a promise.
    if (options.isAsync) {
       return postcss(plugins).process(source, postCSSOptions)
        .then(result => ({
          code: result.css || '',
          map: result.map && (result.map.toJSON() as any),
          errors,
          modules: cssModules,
          rawResult: result,
          dependencies: recordPlainCssDependencies(result.messages)
        }))
        .catch(error => ({
          code: '',
          map: undefined,
          errors: [...errors, error],
          rawResult: undefined,
          dependencies
        }))
    } else {
      result = postcss(plugins).process(source, postCSSOptions).sync()
    }
    recordPlainCssDependencies(result.messages)
    // force synchronous transform (we know we only have sync plugins)
    code = result.css
    outMap = result.map
  } catch (e) {
    errors.push(e)
  }

  return {
    code: code || ``,
    map: outMap && (outMap.toJSON() as any),
    errors,
    rawResult: result,
    dependencies
  }
```

This will break several tests, after switching back to `LazyResult` everything is working as expected.

Reply:
Yes, it looks like one would need to refactor some of the code using the compiler itself, but that might not be useful as the latest commits on the master branch affect `sfc-compiler` quite heavily, which could mean even further refactorings are coming. It's probably not useful to go beyond the minimal changes to make it work with PostCSS 8, unless you can find out where the maintainers want to go.

Reply:
I totally agree with you. In this case it's better to keep it like it is for the moment ;)